app/models/manageiq/providers/redhat/infra_manager/vm/reconfigure.rb

```diff
@@ -27,7 +27,32 @@ def build_config_spec(task_options)
     {
       "numCoresPerSocket" => (task_options[:cores_per_socket].to_i if task_options[:cores_per_socket]),
       "memoryMB"          => (task_options[:vm_memory].to_i if task_options[:vm_memory]),
-      "numCPUs"           => (task_options[:number_of_cpus].to_i if task_options[:number_of_cpus])
+      "numCPUs"           => (task_options[:number_of_cpus].to_i if task_options[:number_of_cpus]),
+      "disksRemove"       => (spec_for_removed_disks(task_options[:disk_remove]) if task_options[:disk_remove]),
+      "disksAdd"          => (spec_for_added_disks(task_options[:disk_add]) if task_options[:disk_add])
     }
   end
+
+  def spec_for_removed_disks(disks)
+    disks.map { |disk| disk["disk_name"] }
+  end
+
+  def spec_for_added_disks(disks)
+    disks.each { |disk_spec| disk_spec["format"] = disk_format_for(disk_spec["thin_provisioned"]) }
+    {
+      "disks"           => disks,
+      "ems_storage_uid" => ManageIQ::Providers::Redhat::InfraManager.extract_ems_ref_id(storage.ems_ref),
+    }
+  end
+
+  FILE_STORAGE_TYPE = %w(NFS GLUSTERFS VMFS).freeze
+  BLOCK_STORAGE_TYPE = %w(FCP ISCSI).freeze
+
+  def disk_format_for(thin_provisioned)
```

Comment:
According to the API it is a required parameter and the request will fail without it. However, if the storage type is different from the examined (non FS and non block) - meaning this is a storage type which isn't categorized and it should be considered here as well.

I'll change the default return value to be 'raw' which should be supported by all.